### PR TITLE
Porting catalog queries to ICG

### PR DESCRIPTION
**Pull request by vraghavan78**
Porting catalog queries tests from TINC_BFV.

Kind Attention: @hlinnaka @danielgustafsson @xinzweb @oarap @hsyuan

### CODE REVIEW

File: src/test/regress/sql/bfv_catalog.sql [hunk 1]
```diff
@@ -527,6 +527,52 @@ from
...
     konst.dim_bezei     = 'NatCat-Zone'   and gesamt.dim_bezei    = 'NatCat-Zone'   and gesamt.dim_elem_lvl = 0;
 
+-- test queries that should run on the master
+drop table if exists mpp_bfv_1;
+create table mpp_bfv_1(col1 int, col2 text, col3 numeric) distributed by (col1);
+
+-- this cannot go to the _setup file, because it is not propagated here
+-- start_ignore
+select disable_xform('CXformIndexGet2IndexScan');
+set optimizer_enable_master_only_queries = on;
```
Comment (hsyuan) [inline src/test/regress/sql/bfv_catalog.sql]: Don't forget to reset this GUC after the query was done.

Comment (vraghavan78) [inline src/test/regress/sql/bfv_catalog.sql]: This is for the whole test

**Comment by hsyuan** [comment]
LGTM.

**Comment by vraghavan78** [comment]
Thanks!

**Pull request closed without merging**